### PR TITLE
DR-555 snapshot file endpoints

### PR DESCRIPTION
**Pull request by ddietterich**
Add snapshot endpoints and update DRS endpoints.
A few things to note:
1. I added FSContainerInterface to give the filesystem code a common way to handle lookups from dataset and snapshot. Maybe too small a convenience for adding an interface?
2. DRS ID is changed. It now contains _only_ the snapshot id and file id. We don't need the dataset id, because we look the file up in the snapshot directory. Yay!

### CODE REVIEW

File: src/main/java/bio/terra/filesystem/FireStoreObject.java [hunk 1]
```diff
@@ -25,12 +25,14 @@
     private String name; // name of the object
 
     // fileref-only fields
-    private String datasetId;
+    // FireStoreFile object in the files collection
+    private String datasetId; // The pari (datasetId, objectId) are used to lookup the FireStoreFile in file collection.
```
Comment (thathert) [inline src/main/java/bio/terra/filesystem/FireStoreObject.java]: pari -> pair

File: src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java [hunk 2]
```diff
@@ -966,8 +964,6 @@ private void buildSourceSelectList(StringBuilder builder,
                     builder.append("CONCAT('drs://")
                         .append(datarepoDnsName)
                         .append("/v1_")
-                        .append(source.getDataset().getId().toString())
```
Comment (thathert) [inline src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java:969]: :)

File: src/main/java/bio/terra/service/DrsService.java [hunk 3]
```diff
@@ -208,6 +206,12 @@ private String makeHttpsFromGs(String gspath) {
     }
 
     private List<String> makeAuthHeader(AuthenticatedUserRequest authUser) {
+        // TODO: I added this so that connected tests would work. Seems like we should have a better solution.
```
Comment (thathert) [inline src/main/java/bio/terra/service/DrsService.java:216]: I created [DR-562](https://broadworkbench.atlassian.net/browse/DR-562) for this.

File: src/test/java/bio/terra/filesystem/EncodeFileTest.java [hunk 4]
```diff
@@ -150,6 +179,107 @@ public void encodeFileTest() throws Exception {
...
+    private void testSnapEnum(Map<String, List<String>> dirmap,
+                              String snapshotId,
+                              String datasetPath,
+                              int inDepth) throws Exception {
+        logger.info(">> testsnapEnum with depth = " + inDepth);
+        FSObjectModel fsObj = connectedOperations.lookupSnapshotFileByPath(snapshotId, datasetPath, inDepth);
+        int maxDepth = checkSnapEnum(dirmap, 0, fsObj);
+        int depth = (inDepth == -1) ? 7 : inDepth;
+        logger.info(">> testSnapEnum maxDepth = " + maxDepth + "  compareDepth = " + depth);
+//        assertThat("Depth is correct", maxDepth, equalTo(depth));
```
Comment (thathert) [inline src/test/java/bio/terra/filesystem/EncodeFileTest.java]: Should this be commented out?

File: src/test/java/bio/terra/filesystem/EncodeFileTest.java [hunk 5]
```diff
@@ -150,6 +179,107 @@ public void encodeFileTest() throws Exception {
...
+            .collect(Collectors.toList());
+
+        // lookup the dirmap list by path of the fsObj
+        List<String> mapList = dirmap.get(fsObj.getPath());
+
+        logger.info(">>>> contents: " + StringUtils.join(contentsNames, ","));
+        logger.info(">>>> mapList : " + StringUtils.join(mapList, ","));
+        // compare the lists
+        StringListCompare slc = new StringListCompare(contentsNames, mapList);
+//        assertTrue("Directory contents match", slc.compare());
```
Comment (thathert) [inline src/test/java/bio/terra/filesystem/EncodeFileTest.java]: And this?

**Review by thathert — APPROVED** [review]
Looks good -- a couple questions about commented out test code at the end.

**Comment by thathert** [comment]
Looks like a checkstyles line length issue that caused travis to complain

**Review by RoriCremer — APPROVED** [review]

**Pull request merged**